production/ksonnet/loki/common.libsonnet

```diff
@@ -32,6 +32,20 @@ local k = import 'ksonnet-util/kausal.libsonnet';
     // This helps ensure we create SRV records starting with _grpclb
     grpclbServiceFor(deployment):: k.util.serviceFor(deployment, $._config.service_ignored_labels, nameFormat='%(port)s'),
 
+    // A headless service for discovering IPs of each query-frontend/scheduler pod.
+    // It leaves it up to the client to do any load-balancing of requests.
+    headlessService(deployment, name)::
+      $.grpclbServiceFor(deployment) +
+      // Make sure that query frontend/scheduler worker, running in the querier, do resolve
+      // each query-frontend/scheduler pod IP and NOT the service IP. To make it, we do NOT
+      // use the service cluster IP so that when the service DNS is resolved it
+      // returns the set of query-frontend/scheduler IPs.
+      k.service.mixin.spec.withClusterIp('None') +
+      // Query frontend/scheduler will not become ready until at least one querier connects
+      // which creates a chicken and egg scenario if we don't publish the
+      // query-frontend/scheduler address before it's ready.
```

Comment:
I'm not sure these comments are best located here since this is a helper function.
Can you move them somewhere more appropriate pls?